catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/SolrCacheTest.java

```diff
@@ -87,6 +83,55 @@ public void setUp() {
             Collections.singletonList(mockCachePutPlugin));
   }
 
+  /**
+   * Verify that if the CachePutPlugin returns Optional.empty(), then the metacard is not added to
+   * the cache.
+   */
+  @Test
+  public void testUncacheableMetacardsAreNotCached() {
+
+    mockEmptyCachePutPlugin();
+
+    List<Metacard> metacards = new ArrayList<>();
+    Metacard metacard = mock(Metacard.class);
+    when(metacard.getSourceId()).thenReturn(TEST_ID);
+    when(metacard.getId()).thenReturn(TEST_ID);
+    metacards.add(metacard);
+
+    solrCache.put(metacards);
+
+    Mockito.verifyNoMoreInteractions(mockSolrClient, mockCacheSolrMetacardClient);
+  }
+
+  @Test
+  public void testCachePutPluginModifiesMetacard() {
+
+    String name = "attribute";
+    String value = "value";
+
+    List<Metacard> metacards = new ArrayList<>();
+    Metacard metacard = mock(Metacard.class);
+    when(metacard.getSourceId()).thenReturn(TEST_ID);
+    when(metacard.getId()).thenReturn(TEST_ID);
+    metacards.add(metacard);
+
+    when(mockCachePutPlugin.process(any()))
+        .thenAnswer(
+            (Answer<Optional<Metacard>>)
+                invocationOnMock -> {
+                  Metacard metacardToModify = (Metacard) invocationOnMock.getArguments()[0];
+                  metacardToModify.setAttribute(new AttributeImpl(name, value));
+                  return Optional.of(metacardToModify);
+                });
+
+    solrCache.put(metacards);
+
+    ArgumentCaptor<Attribute> argumentCaptor = ArgumentCaptor.forClass(Attribute.class);
```

Comment:
✏️ I was thinking we could use a captor for the metacard list in the `metacardClient.add(updatedMetacards, false)` call to verify that the modified metacards are actually passed to the cache client. Could use a `MetacardImpl` instead of a mock to make the verification easier too.

Reply:
done